feat: emoji picker improvement

hyj1204 opened the pull request:
<!---
Thank you for submitting a pull request to AppFlowy. The team will dedicate their best efforts to reviewing and approving your pull request. If you have any questions about the project or feedback for us, please join our [Discord](https://discord.gg/wdjWUXXhtw).
-->

<!---
If your pull request adds a new feature, please drag and drop a video into this section to showcase what you've done! If not, you may delete this section.
-->

### Feature Preview
This PR is to refactor and improve the UI of the emoji picker. 
1. Delete the duplicate code under `appflowy_flutter/lib/plugins/document/presentation/editor_plugins/emoji_picker` . 
The emoji picker related code is only under `appflowy_flutter/lib/workspace/presentation/widgets/emoji_picker`. 

**The following files were impacted by this change.** 
RowBanner: 
`appflowy_flutter/lib/plugins/database_view/widgets/row/row_banner.dart`
EmojiPickerButton: `appflowy_flutter/lib/plugins/document/presentation/editor_plugins/base/emoji_picker_button.dart`
DocumentHeaderNodeWidget(Add icon in header): `appflowy_flutter/lib/plugins/document/presentation/editor_plugins/header/document_header_node_widget.dart`
EmojiPopover(The pop menu on `DocumentHeaderNodeWidget`): `appflowy_flutter/lib/plugins/document/presentation/editor_plugins/header/emoji_popover.dart`
AppFlowyEditorPage(slash menu):
`appflowy_flutter/lib/plugins/document/presentation/editor_page.dart`



2. Add `buildFlowyEmojiPickerConfig` method to unify the UI style of the emoji picker we used in `EmojiPopover`(on the Header) and `EmojiSelectionMenu`(on slash menu)

3. Improve and refactor
- Improve UI for DefaultEmojiPickerView
- Improve EmojiPickerConfig
- Add hover effect on the emoji button
- Improve parameters in EmojiPickerConfig
- Add LocaleKey for prompt texts

Light mode:



https://github.com/AppFlowy-IO/AppFlowy/assets/14248245/b9f751d1-c330-48e8-babd-483949d5e2ab

Dark mode:



https://github.com/AppFlowy-IO/AppFlowy/assets/14248245/4e131628-5ce8-4c51-bd2a-e68e7bab57fd





<!---
List at least one issue here that this PR addresses. If it fixes the issue, please use the [fixes](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests) keyword to close the issue. For example:
fixes https://github.com/AppFlowy-IO/AppFlowy/pull/2106
-->

---

<!---
Before you mark this PR ready for review, run through this checklist!
-->

#### PR Checklist

- [x] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [x] I've listed at least one issue that this PR fixes in the description above.
- [x] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [x] All existing tests are passing.

rituraj2000 commented:
Please assign it to me @LucasXu0 @Xazin

Xazin commented:
> Please assign it to me @LucasXu0 @Xazin 

Hey Rituraj, this is not an unassigned issue, this is a pull request.

You can check here for issues: https://github.com/AppFlowy-IO/AppFlowy/issues

If you're looking for unassigned issues and unassigned hacktober issues, look here:

https://github.com/AppFlowy-IO/AppFlowy/issues?q=is%3Aissue+is%3Aopen+no%3Aassignee

https://github.com/AppFlowy-IO/AppFlowy/issues?q=is%3Aissue+is%3Aopen+no%3Aassignee+label%3Ahacktoberfest

richardshiue commented:
LGTM.

What was the reason behind not using freezed for the EmojiPickerConfig initially?

Is it possible to keep the version in workspace and remove the one in editor plugins instead? The emoji picker is used in other parts of AppFlowy as well.

Xazin commented on frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/emoji_picker/src/default_emoji_picker_view.dart:
```diff
@@ -278,14 +270,16 @@ class DefaultEmojiPickerViewState extends State<DefaultEmojiPickerView>
+          fit: BoxFit.scaleDown,
+          child: Text(
+            emoji.emoji,
+            textScaleFactor: 1.0,
```
Do we need to force `textScaleFactor: 1.0` here?

hyj1204 replied:
No, I think it is unnecessary. I also removed the `BoxFit`, the default value `BoxFit.contain` should be enough here.

```dart
   child: FlowyHover(
        child: FittedBox(
          child: Text(
            emoji.emoji,
            style: TextStyle(
              fontSize: emojiSize,
            ),
          ),
        ),
      ),
```

Xazin commented on frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/emoji_picker/src/flowy_emoji_picker_config.dart:
```diff
@@ -0,0 +1,32 @@
+    selectedCategoryIconBackgroundColor: style.colorScheme.primary,
+    progressIndicatorColor: style.colorScheme.primary,
+    backspaceColor: style.colorScheme.primary,
+    searchHintText: 'Search emoji localations',
+    serachHintTextStyle: style.textTheme.bodyMedium?.copyWith(
+      color: style.hintColor,
+    ),
+    serachBarEnableBorder: OutlineInputBorder(
+      borderRadius: BorderRadius.circular(4),
+      borderSide: BorderSide(color: style.dividerColor),
+    ),
+    serachBarFocusedBorder: OutlineInputBorder(
+      borderRadius: BorderRadius.circular(4),
+      borderSide: BorderSide(
+        color: style.colorScheme.primary,
+      ),
+    ),
+    noRecentsText: 'No Recents localations',
+    noRecentsStyle: style.textTheme.bodyMedium,
+    noEmojiFoundText: 'No emoji found localations',
```
Localization

Xazin commented on frontend/appflowy_flutter/packages/flowy_infra_ui/lib/style_widget/scrolling/styled_scroll_bar.dart:
```diff
@@ -225,6 +225,7 @@ class ScrollbarListStack extends StatelessWidget {
   final EdgeInsets? scrollbarPadding;
   final Color? handleColor;
   final Color? trackColor;
+  final bool showTrack;
```
You added this property, but don't set it to `true` anywhere, and it defaults to `false` inside `StyledScrollbar`, do we need it?

hyj1204 replied:
I added it because `ScrollbarListStack` has `showTrack` parameter, but it didn't pass down to `StyledScrollbar` in the previous code. The previous UI didn't show track, so I didn't set `true` there. 
However, I think it would be better to show the track there. 
I also updated the default color of track like this: 
Light mode: 
<img width="148" alt="image" src="https://github.com/AppFlowy-IO/AppFlowy/assets/14248245/8f03f9fc-4e19-4f96-8242-d10d810d7e59">

Dark mode:
<img width="133" alt="image" src="https://github.com/AppFlowy-IO/AppFlowy/assets/14248245/4346050e-9c9b-4930-accc-8730a797f9d4">

hyj1204 commented:
> LGTM.
> 
> What was the reason behind not using freezed for the EmojiPickerConfig initially?
> 
> Is it possible to keep the version in workspace and remove the one in editor plugins instead? The emoji picker is used in other parts of AppFlowy as well. (like row_banner.dart but I think I wrote some bad code that imports editor plugin version...)

I didn't know the reason and it seems no problem using freezed here, so I updated it with freezed now.

I also moved the emoji picker part to the workspace(PR description has been updated too) since it was used in multiple places. I totally agree with you.

hyj1204 commented on frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/emoji_picker/src/emji_picker_config.dart:
```diff
@@ -6,32 +6,37 @@ import 'models/category_models.dart';
     this.emojiSizeMax = 32.0,
     this.verticalSpacing = 0,
     this.horizontalSpacing = 0,
     this.initCategory = Category.RECENT,
```
I agree with you. I just updated some parameters' names to make it more readable.

Category ->  EmojiCategory

CategoryEmoji -> EmojiCategoryGroup (EmojiCategory with its emojis)

CategoryIcon ->  EmojiCategoryIcon

Xazin replied:
Looks good to me :+1:
